### PR TITLE
Add the names flag for pod logs

### PR DESCRIPTION
**Pull request by keonchennl**
Fixes containers#13261

Signed-off-by: Xueyuan Chen <X.Chen-47@student.tudelft.nl>

### CODE REVIEW

File: test/e2e/logs_test.go [hunk 1]
```diff
@@ -37,7 +37,10 @@ var _ = Describe("Podman logs", func() {
 		}
 		podmanTest = PodmanTestCreate(tempdir)
 		podmanTest.Setup()
-		podmanTest.SeedImages()
+		err := podmanTest.SeedImages()
```
Comment (rhatdan) [inline test/e2e/logs_test.go:40]: Nit: Please write this as 
```
if err := podmanTest.SeedImages(); err != nil {
}
```

File: test/e2e/logs_test.go [hunk 2]
```diff
@@ -412,4 +415,30 @@ var _ = Describe("Podman logs", func() {
...
+		log1.WaitWithDefaultTimeout()
+		Expect(log1).To(Exit(0))
+
+		log2 := podmanTest.Podman([]string{"run", "--name", containerName2, "-d", "--pod", podName, BB, "/bin/sh", "-c", "echo podman"})
+		log2.WaitWithDefaultTimeout()
+		Expect(log2).To(Exit(0))
+
+		results := podmanTest.Podman([]string{"pod", "logs", "-names", podName})
+		results.WaitWithDefaultTimeout()
+		Expect(results).To(Not(Exit(0)))
```
Comment (rhatdan) [inline test/e2e/logs_test.go:438]: Could you check this to Exit(125)

File: test/e2e/logs_test.go [hunk 3]
```diff
@@ -412,4 +415,30 @@ var _ = Describe("Podman logs", func() {
...
+
+		output := results.OutputToStringArray()
+		Expect(output).To(HaveLen(2))
+		Expect(strings.Contains(output[0], containerName1) && strings.Contains(output[0], containerName2)).To(BeTrue())
```
Comment (rhatdan) [inline test/e2e/logs_test.go:442]: Expect(output).To(ContainSubstring(containerName1))
Expect(output).To(ContainSubstring(containerName2))

**Review by rhatdan — APPROVED** [review]
Thanks @keonchennl 
A couple of Nits

File: test/e2e/logs_test.go (hunk 2 above)
Comment (keonchennl) [reply test/e2e/logs_test.go]: Sorry I made a mistake. Here it should exit without error. I fixed it to expecting `Exit(0)`.

**Comment by keonchennl** [comment]
Hi Daniel @rhatdan I am sort of stuck in fixing the pipeline for this MR:
My local test run passes (on ubuntu 20.04) and manual testing also gave the correct result: the container names show up if I run  `podman pod logs --names mypod`.
But the integration tests in the pipeline keep failing in this test case: the container names are even gone with the added "--names" flag.
Could you help have a look what went wrong? 

Below is the result of one of the integration tests (int-podman-ubuntu-2110-root-host)

         [+0387s] Podman logs 
         [  podman pod logs with container names]()
           /var/tmp/go/src/github.com[/containers/podman/test/e2e/logs_test.go:418](https://github.com/containers/podman/blob/9990172cb0e671e39830db3161a1b58fe984468f/test/e2e/logs_test.go#L418)
         
         [BeforeEach] Podman logs
           /var/tmp/go/src/github.com[/containers/podman/test/e2e/logs_test.go:33](https://github.com/containers/podman/blob/9990172cb0e671e39830db3161a1b58fe984468f/test/e2e/logs_test.go#L33)
         [It] podman pod logs with container names
           /var/tmp/go/src/github.com[/containers/podman/test/e2e/logs_test.go:418](https://github.com/containers/podman/blob/9990172cb0e671e39830db3161a1b58fe984468f/test/e2e/logs_test.go#L418)
         # podman [options] --network-backend cni --storage-driver vfs pod create --name=test-pod
         24926f1ad6c6408d1a77904b2d8a84e5f66f34dfe166a7698bdb95230b5962be
         # podman [options] --network-backend cni --storage-driver vfs run --name container-1 -d --pod test-pod quay.io/libpod/busybox:latest /bin/sh -c echo podman
         7c3a11e95095aa573609b5dcb08f9c85a3f62a3eaa1b2e2ce1398361976a740a
         # podman [options] --network-backend cni --storage-driver vfs run --name container-2 -d --pod test-pod quay.io/libpod/busybox:latest /bin/sh -c echo podman
         72a56649efdeb29091f58bb1328b2d4684600ca3038338c405160bdae4698078
         # podman [options] --network-backend cni --storage-driver vfs pod logs --names test-pod
          podman
          podman
         [AfterEach] Podman logs
           /var/tmp/go/src/github.com[/containers/podman/test/e2e/logs_test.go:45](https://github.com/containers/podman/blob/9990172cb0e671e39830db3161a1b58fe984468f/test/e2e/logs_test.go#L45)
         # podman [options] --network-backend cni --storage-driver vfs stop -a --time 0
         72a56649efdeb29091f58bb1328b2d4684600ca3038338c405160bdae4698078
         7c3a11e95095aa573609b5dcb08f9c85a3f62a3eaa1b2e2ce1398361976a740a
         afa2f8ffd9014cfe396c3d7cbe222a73aac5e81b14c1ab896f7a40ae1f66336a
         # podman [options] --network-backend cni --storage-driver vfs pod stop -a -t 0
         24926f1ad6c6408d1a77904b2d8a84e5f66f34dfe166a7698bdb95230b5962be
         # podman [options] --network-backend cni --storage-driver vfs pod rm -fa
         24926f1ad6c6408d1a77904b2d8a84e5f66f34dfe166a7698bdb95230b5962be
         # podman [options] --network-backend cni --storage-driver vfs rm -fa

**Comment by rhatdan** [comment]
Add this patch 
[patch.txt](https://github.com/containers/podman/files/8155951/patch.txt)

File: docs/source/markdown/podman-pod-logs.1.md [hunk 4]
```diff
@@ -28,6 +28,10 @@ chance that the log file will be removed before `podman pod logs` reads the fina
 
 Instead of providing the pod name or id, get logs of the last created pod. (This option is not available with the remote Podman client, including Mac and Windows (excluding WSL2) machines)
 
+#### **--names**, **-n**
+
+Output the container names instead of the container IDs in the log
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-pod-logs.1.md:33]:
```suggestion
Output the container names instead of the container IDs in the log.
```

**Comment by rhatdan** [comment]
LGTM other then @TomSweeneyRedHat nit.

**Comment by TomSweeneyRedHat** [comment]
LGTM other than that nit of a nit

File: docs/source/markdown/podman-pod-logs.1.md (hunk 4 above)
Comment (keonchennl) [reply docs/source/markdown/podman-pod-logs.1.md]: Thanks! Fixed

**Comment by keonchennl** [comment]
It seems some integration test cases get unstable. I'm not sure if the changes cause this. @rhatdan 

In last run:

This one was flaky:
> [Fail] podman system service verify pprof endpoints [It] are not available

This one failed in int-podman-ubuntu-2110-rootless-host:
> [Fail] Podman run with --mac-address flag [It] Podman run --mac-address with custom network

**Pull request merged**